### ISSUE TITLE
Thought: Expand feature file includes into .glyphs file?

### ISSUE THREAD

**Issue by madig** [googlei18n/glyphsLib#506]
While preparing Cantarell to use a shared `features.fea`, I hit the snag that if I use a shared file and make the UFO's `features.fea` be just `include(family.fea)`, that's what is put into the feature dialog, preventing simple code updating and export. So I'm thinking about reverting it back to identical copies in each UFO.

Might it be a good idea to magically expand feature files into the .glyphs file if all they have is a single include statement? Or is it a horrible corner-case for papering over an impedance mismatch to the UFO format? Thoughts?

**Comment by schriftgestalt** [comment]
Glyphs expands the path on export. 

When you convert between .glyphs and .ufo, you might need to adjust the (relativ) path depending on the location of the files. And then the feature Compiler can find the .fea file.

**Comment by madig** [comment]
Oh? So if I leave the .fea file next to the .glyphs file, Glyphs can find it, update and export it?

**Comment by schriftgestalt** [comment]
Glyphs supports relativ and absolute paths. Relative paths are replaced by absolute paths on export that makeOTF can find them.

**Comment by madig** [comment]
If I have a `family.fea` next to the `.glyphs` file and make a prefix with just `include(family.fea)`, I get a makeOTF error.

**Comment by schriftgestalt** [comment]
What version?

**Comment by madig** [comment]
2.6. You can test this with the .zip I sent you, extract all features into one file next to the .glyphs file and try it.

**Comment by schriftgestalt** [comment]
You need to put the import into a Prefix, not in a Feature.
Can you send me what you did? I tried it and it worked fine.

**Comment by madig** [comment]
Here: [CantarellExternalFeatures.zip](https://github.com/googlei18n/glyphsLib/files/2873477/CantarellExternalFeatures.zip)

**Comment by madig** [comment]
I found out why it doesn't work: makeOTF expects a semicolon after an include statement, feaLib does not, and I did not have one :)

**Comment by anthrotype** [comment]
Right now, exporting master UFOs + DesignSpace from a .glyphs file with glyphsLib will leave any `include` statements that are present in the original .glyphs source in a broken state if the destination folder where the UFOs are saved is different from the one where the input .glyphs file is located.

This is problem surfaced with fontmake, since the latter by default writes UFOs + DS into a `master_ufo` directory one level down (unless one uses the relative obscure option `--master-dir` to change that).

Basically any .glyphs source with an include statement with relative path will make fontmake crash (https://github.com/googlefonts/fontmake/issues/359).

I think glyphsLib should find/replace the include's relative paths that are present in _either_ a .glyphs input (when exporting to DS) or master UFOs input (when exporting back to .glyphs), and _do the right thing_: i.e. adjust the relative paths according to the destination directory.

**Comment by anthrotype** [comment]
absolute paths are not a good idea, if we want to keep the source files portable (glyphsLib converts between source formats after all). Expanding the features is also not ideal, since it will not round-trip. Only solution I see is to update the relative paths according to the output directory.
